oteapi/datacache/datacache.py

```diff
@@ -109,10 +109,11 @@ def __init__(
             cache_dir = self.config.cacheDir
         if isinstance(cache_dir, str):
             cache_dir = Path(cache_dir)
-        if cache_dir.is_absolute():
+        if cache_dir.is_absolute():  # type: ignore[union-attr]
             self.cache_dir = cache_dir
         else:
-            self.cache_dir = Path(tempfile.gettempdir()).resolve() / cache_dir
+            temp_path = Path(tempfile.gettempdir()).resolve()
+            self.cache_dir = temp_path / cache_dir  # type: ignore[operator]
```

Comment:
Is there a reason to split these lines up in two? If not, please return them to a single line.

Reply:
The problem was that mypy claimed that `cache_dir` was `None`, and thus that the `/` operator was invalid. When I added the ignore statement to the end of the line, it became too long (a different complaint), so I split it up.

Reply:
Yes, it's because mypy thinks `cache_dir` is still either `None` or `Path`, and `None` doesn't have any of these functionalities. Hmm - it's weird. But fair.